<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>mainwindow_view</class>
 <widget class="QMainWindow" name="mainwindow_view">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>800</width>
    <height>600</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string notr="true">CSP</string>
  </property>
  <property name="dockNestingEnabled">
   <bool>true</bool>
  </property>
  <property name="dockOptions">
   <set>QMainWindow::AllowNestedDocks|QMainWindow::AllowTabbedDocks|QMainWindow::AnimatedDocks|QMainWindow::VerticalTabs</set>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QHBoxLayout" name="horizontalLayout">
    <property name="spacing">
     <number>0</number>
    </property>
    <property name="leftMargin">
     <number>0</number>
    </property>
    <property name="topMargin">
     <number>0</number>
    </property>
    <property name="rightMargin">
     <number>0</number>
    </property>
    <property name="bottomMargin">
     <number>0</number>
    </property>
    <item>
     <widget class="QStackedWidget" name="stackedwidget">
      <property name="currentIndex">
       <number>1</number>
      </property>
      <widget class="home_view" name="page_home_view"/>
      <widget class="chip_configure_view" name="page_chip_configure_view"/>
      <widget class="QWidget" name="page_empty"/>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>800</width>
     <height>20</height>
    </rect>
   </property>
   <widget class="QMenu" name="menu">
    <property name="title">
     <string>File</string>
    </property>
    <widget class="QMenu" name="menu_new">
     <property name="title">
      <string>New Project</string>
     </property>
     <addaction name="action_new_chip"/>
    </widget>
    <addaction name="menu_new"/>
    <addaction name="action_load"/>
    <addaction name="action_import"/>
    <addaction name="action_save"/>
    <addaction name="action_saveas"/>
    <addaction name="action_close"/>
    <addaction name="action_report"/>
    <addaction name="separator"/>
   </widget>
   <widget class="QMenu" name="menuProject">
    <property name="title">
     <string>Project</string>
    </property>
    <addaction name="separator"/>
    <addaction name="action_generate"/>
    <addaction name="action_package_manager"/>
   </widget>
   <widget class="QMenu" name="menuHelp">
    <property name="title">
     <string>Help</string>
    </property>
   </widget>
   <addaction name="menu"/>
   <addaction name="menuProject"/>
   <addaction name="menuHelp"/>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
  <widget class="QDockWidget" name="dockwidget_left">
   <property name="minimumSize">
    <size>
     <width>250</width>
     <height>156</height>
    </size>
   </property>
   <property name="features">
    <set>QDockWidget::DockWidgetFloatable|QDockWidget::DockWidgetMovable</set>
   </property>
   <property name="windowTitle">
    <string>Project</string>
   </property>
   <attribute name="dockWidgetArea">
    <number>1</number>
   </attribute>
   <widget class="QWidget" name="dockwidget_left_contents">
    <layout class="QGridLayout" name="gridLayout">
     <property name="leftMargin">
      <number>9</number>
     </property>
     <property name="topMargin">
      <number>0</number>
     </property>
     <property name="rightMargin">
      <number>0</number>
     </property>
     <property name="bottomMargin">
      <number>0</number>
     </property>
     <property name="spacing">
      <number>0</number>
     </property>
     <item row="0" column="0">
      <widget class="QTreeView" name="treeview"/>
     </item>
    </layout>
   </widget>
  </widget>
  <widget class="QDockWidget" name="dockwidget_right">
   <property name="minimumSize">
    <size>
     <width>250</width>
     <height>58</height>
    </size>
   </property>
   <property name="features">
    <set>QDockWidget::DockWidgetFloatable|QDockWidget::DockWidgetMovable</set>
   </property>
   <property name="windowTitle">
    <string>Property</string>
   </property>
   <attribute name="dockWidgetArea">
    <number>2</number>
   </attribute>
   <widget class="QWidget" name="dockwidget_right_contents">
    <layout class="QGridLayout" name="gridLayout_2">
     <property name="leftMargin">
      <number>0</number>
     </property>
     <property name="topMargin">
      <number>0</number>
     </property>
     <property name="rightMargin">
      <number>9</number>
     </property>
     <property name="bottomMargin">
      <number>0</number>
     </property>
     <property name="spacing">
      <number>0</number>
     </property>
     <item row="0" column="0">
      <widget class="propertybrowser" name="treepropertybrowser" native="true"/>
     </item>
    </layout>
   </widget>
  </widget>
  <widget class="QDockWidget" name="dockwidget_bottom_output">
   <property name="minimumSize">
    <size>
     <width>88</width>
     <height>107</height>
    </size>
   </property>
   <property name="features">
    <set>QDockWidget::DockWidgetFloatable|QDockWidget::DockWidgetMovable</set>
   </property>
   <property name="allowedAreas">
    <set>Qt::AllDockWidgetAreas</set>
   </property>
   <property name="windowTitle">
    <string>Output</string>
   </property>
   <attribute name="dockWidgetArea">
    <number>8</number>
   </attribute>
   <widget class="QWidget" name="dockwidget_bottom_output_contents">
    <layout class="QGridLayout" name="gridLayout_3">
     <item row="0" column="0">
      <widget class="logviewbox" name="logviewbox_output">
       <property name="readOnly">
        <bool>true</bool>
       </property>
      </widget>
     </item>
    </layout>
   </widget>
  </widget>
  <widget class="QDockWidget" name="dockwidget_bottom_configurations">
   <property name="features">
    <set>QDockWidget::DockWidgetFloatable|QDockWidget::DockWidgetMovable</set>
   </property>
   <property name="windowTitle">
    <string>Configurations</string>
   </property>
   <attribute name="dockWidgetArea">
    <number>8</number>
   </attribute>
   <widget class="QWidget" name="dockwidget_bottom_configurations_contents"/>
  </widget>
  <widget class="QDockWidget" name="dockwidget_bottom_xmake_output">
   <property name="features">
    <set>QDockWidget::DockWidgetFloatable|QDockWidget::DockWidgetMovable</set>
   </property>
   <property name="windowTitle">
    <string>Xmake Output</string>
   </property>
   <attribute name="dockWidgetArea">
    <number>8</number>
   </attribute>
   <widget class="QWidget" name="dockwidget_bottom_xmake_output_contents">
    <layout class="QGridLayout" name="gridLayout_4">
     <item row="0" column="0">
      <widget class="logviewbox" name="logviewbox_xmake_output">
       <property name="readOnly">
        <bool>true</bool>
       </property>
      </widget>
     </item>
    </layout>
   </widget>
  </widget>
  <widget class="QToolBar" name="toolbar">
   <property name="windowTitle">
    <string>toolBar</string>
   </property>
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
   <addaction name="action_generate"/>
   <addaction name="action_package_manager"/>
  </widget>
  <action name="action_load">
   <property name="text">
    <string>Load Project</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+L</string>
   </property>
  </action>
  <action name="action_import">
   <property name="text">
    <string>Import Project</string>
   </property>
  </action>
  <action name="action_save">
   <property name="text">
    <string>Save Project</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+S</string>
   </property>
  </action>
  <action name="action_saveas">
   <property name="text">
    <string>Save Project As ...</string>
   </property>
  </action>
  <action name="action_close">
   <property name="text">
    <string>Close Project</string>
   </property>
  </action>
  <action name="action_report">
   <property name="text">
    <string>Generate Report</string>
   </property>
  </action>
  <action name="action_new_chip">
   <property name="text">
    <string>New Chip Project</string>
   </property>
  </action>
  <action name="action_generate">
   <property name="icon">
    <iconset resource="../resources/resources.qrc">
     <normaloff>:/icon/code-s-slash-line.svg</normaloff>:/icon/code-s-slash-line.svg</iconset>
   </property>
   <property name="text">
    <string>Generate code</string>
   </property>
  </action>
  <action name="action_package_manager">
   <property name="icon">
    <iconset resource="../resources/resources.qrc">
     <normaloff>:/icon/dropbox-line.svg</normaloff>:/icon/dropbox-line.svg</iconset>
   </property>
   <property name="text">
    <string>Package Manager</string>
   </property>
  </action>
 </widget>
 <customwidgets>
  <customwidget>
   <class>home_view</class>
   <extends>QWidget</extends>
   <header location="global">home_view.h</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>chip_configure_view</class>
   <extends>QWidget</extends>
   <header location="global">chip_configure_view.h</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>propertybrowser</class>
   <extends>QWidget</extends>
   <header>propertybrowser.h</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>logviewbox</class>
   <extends>QTextEdit</extends>
   <header>logviewbox.h</header>
  </customwidget>
 </customwidgets>
 <resources>
  <include location="../resources/resources.qrc"/>
 </resources>
 <connections/>
</ui>
